### PR TITLE
refactor: cleanup main.rs

### PR DESCRIPTION
**Pull request by dsherret**
1. Extracts out some code from main.rs
2. Inlines all the `x_command` functions in main.rs

### CODE REVIEW

File: cli/graph_util.rs [hunk 1]
```diff
@@ -496,3 +506,105 @@ pub fn graph_lock_or_exit(graph: &ModuleGraph, lockfile: &mut Lockfile) {
     }
   }
 }
+
+pub async fn create_graph_and_maybe_check(
```
Comment (dsherret) [inline cli/graph_util.rs:510]: Straight function extracts. No changes within them.

File: cli/main.rs [hunk 2]
```diff
@@ -776,74 +68,140 @@ fn init_v8_flags(v8_flags: &[String]) {
...
     }
     DenoSubcommand::Task(task_flags) => {
-      task_command(flags, task_flags).boxed_local()
+      tools::task::execute_script(flags, task_flags).await
     }
     DenoSubcommand::Test(test_flags) => {
-      test_command(flags, test_flags).boxed_local()
+      if let Some(ref coverage_dir) = flags.coverage_dir {
+        std::fs::create_dir_all(coverage_dir)
+          .with_context(|| format!("Failed creating: {}", coverage_dir))?;
```
Comment (dsherret) [inline cli/main.rs:171]: Added this with_context.

File: cli/main.rs [hunk 3]
```diff
@@ -776,74 +68,140 @@ fn init_v8_flags(v8_flags: &[String]) {
...
+        // this is set in order to ensure spawned processes use the same
+        // coverage directory
```
Comment (dsherret) [inline cli/main.rs:173]: Added this comment.

File: cli/main.rs [hunk 4]
```diff
@@ -776,74 +68,140 @@ fn init_v8_flags(v8_flags: &[String]) {
   }
 }
 
-fn get_subcommand(
-  flags: Flags,
-) -> Pin<Box<dyn Future<Output = Result<i32, AnyError>>>> {
```
Comment (dsherret) [inline cli/main.rs:781]: It seems this boxing was unnecessary?

File: cli/proc_state.rs [hunk 5]
```diff
@@ -469,6 +470,30 @@ impl ProcState {
     Ok(())
   }
 
+  /// Helper around prepare_module_load that loads and type checks
+  /// the provided files.
+  pub async fn load_and_type_check_files(
```
Comment (dsherret) [inline cli/proc_state.rs:475]: Extracted out of main.rs

File: cli/tools/bundle.rs [hunk 6]
```diff
@@ -0,0 +1,158 @@
...
+use crate::args::TsConfigType;
+use crate::args::TypeCheckMode;
+use crate::graph_util::create_graph_and_maybe_check;
+use crate::graph_util::error_for_any_npm_specifier;
+use crate::proc_state::ProcState;
+use crate::util;
+use crate::util::display;
+use crate::util::file_watcher::ResolutionResult;
+
+pub async fn bundle(
```
Comment (dsherret) [inline cli/tools/bundle.rs:23]: Extracted out of main.rs. No changes other than some added qualifiers.

File: cli/tools/repl/mod.rs [hunk 7]
```diff
@@ -76,11 +78,17 @@ async fn read_eval_file(
...
+  let main_module = resolve_url_or_path("./$deno$repl.ts").unwrap();
+  let ps = ProcState::build(flags).await?;
+  let mut worker = create_main_worker(
+    &ps,
+    main_module.clone(),
+    Permissions::from_options(&ps.options.permissions_options())?,
+  )
+  .await?;
+  worker.setup_repl().await?;
+  let worker = worker.into_main_worker();
```
Comment (dsherret) [inline cli/tools/repl/mod.rs:91]: Extracted out of main.rs

File: cli/tools/run.rs [hunk 8]
```diff
@@ -0,0 +1,169 @@
...
+use crate::args::EvalFlags;
+use crate::args::Flags;
+use crate::args::RunFlags;
+use crate::file_fetcher::File;
+use crate::npm::NpmPackageReference;
+use crate::proc_state::ProcState;
+use crate::util;
+use crate::worker::create_main_worker;
+
+pub async fn run_script(
```
Comment (dsherret) [inline cli/tools/run.rs:22]: All extracted out of main.rs. No changes.

File: cli/tools/standalone.rs [hunk 9]
```diff
@@ -25,10 +29,59 @@ use std::io::SeekFrom;
 use std::io::Write;
 use std::path::Path;
 use std::path::PathBuf;
+use std::sync::Arc;
 
 use super::installer::infer_name_from_url;
 
-pub async fn get_base_binary(
+pub async fn compile(
```
Comment (dsherret) [inline cli/tools/standalone.rs:36]: Again, also extracted. No changes.

**Comment by wojtekxtx** [comment]
@dsherret by `Inlines` you mean `format`?

**Comment by dsherret** [comment]
> @dsherret by Inlines you mean format? If so, than I agree. Formatting makes code nicer to view, easier to read/maintain.

@wojtekxtx no, I meant inlines, like so, but for functions: https://refactoring.guru/inline-method -- For example, in main.rs `init_command`'s body was inlined into its calling function.

**Review by bartlomieju — APPROVED** [review]
LGTM, nice cleanup

**Pull request merged**